experimental/gcp-log/internal/storage/storage.go

```diff
@@ -405,7 +405,8 @@ func (c *Client) StoreTile(ctx context.Context, level, index uint64, tile *api.T
 	tPath := filepath.Join(layout.TilePath("", level, index, tileSize%256))
 	obj := bkt.Object(tPath)
 
-	w := obj.NewWriter(ctx)
+	// Tiles, partial or full, should only be written once.
+	w := obj.If(gcs.Conditions{DoesNotExist: true}).NewWriter(ctx)
```

Comment:
If the precondition is not met will there be an error at L413?

Reply:
The error will actually be returned in `w.close()`.

See the example in the `If` [documentation](https://pkg.go.dev/cloud.google.com/go/storage#example-ObjectHandle.If) and "Since writes happen asynchronously, Write may return a nil error even though the write failed (or will fail). Always use the error returned from Writer.Close to determine if the upload was successful." from [Write](https://pkg.go.dev/cloud.google.com/go/storage#Writer.Write).

Reply:
Cool, thanks!

Ok, so if there is an existing tile present then I imagine all further updates to the log will be "wedged" until the operator figures out what's gone wrong and fixes the storage (possibly by removing tiles which shouldn't be there if e.g. `integrate` wrote new tiles  but then crashed before the `checkpoint` file was updated)?

Might be worth filing an issue for us to update/add docs for this case, wdyt?

Reply:
As we discussed, I changed the logic here to check that the content is the same if it exists. PTAL!

Reply:
Also, filed the issue here: https://github.com/transparency-dev/serverless-log/issues/72